addons/l10n_fr_account/models/account_move.py

```diff
@@ -20,3 +20,16 @@ def _get_view(self, view_id=None, view_type='form', **options):
     def _compute_l10n_fr_is_company_french(self):
         for record in self:
             record.l10n_fr_is_company_french = record.country_code in record.company_id._get_france_country_codes()
+
+    @api.depends('country_code', 'move_type')
+    def _compute_show_delivery_date(self):
+        super()._compute_show_delivery_date()
+        for move in self:
+            if move.l10n_fr_is_company_french:
+                move.show_delivery_date = move.is_sale_document()
+                
+    def _post(self, soft=True):
+        for move in self:
+            if move.l10_fr_is_company_french and move.is_sale_document() and not move.delivery_date:
+                move.delivery_date = move.invoice_date or fields.Date.context_today(self)
+        return super()._post(soft)
```

Comment:
Care with the super that if in the execution the delivery date is modified then the behavior would be different so i would put the super at the beginning of the function to be extra safe

Reply:
Didn't understand this, if I put super at the beginning what would the function return ?

Reply:
you can do something like:
```suggestion
       post =  super()._post(soft)
       ....
       return post
```